core/src/main/java/org/geysermc/geyser/translator/inventory/PlayerInventoryTranslator.java

```diff
@@ -127,8 +127,8 @@ public void updateSlot(GeyserSession session, Inventory inventory, int slot) {
             // Check for custom skull
             if (javaItem.getJavaId() == session.getItemMappings().getStoredItems().playerHead().getJavaId()
                     && javaItem.getNbt() != null
-                    && javaItem.getNbt().get("SkullOwner") instanceof CompoundTag profile) {
-                FakeHeadProvider.setHead(session, session.getPlayerEntity(), profile);
+                    && javaItem.getNbt().get("SkullOwner") != null) {
```

Comment:
Is there a way we can refactor this so we don't have to call the `get` twice on SkullOwner? Maybe the method in FakeHeadProvider can take the SkullOwner tag response and also check for null?

Reply:
good idea; i'll pass on .get skullowner, that way i'll incorporate the null check and instance of StringTag/CompoundTag with the parameter